yax/shared/utilities/taxidtool.py

```diff
@@ -0,0 +1,445 @@
+    print("Entering output_sequences")
+
+    print_line = False
+    taxid_counts = {}
+    with open(sequences_input_fp, 'r') as sequences_fh:
+        with open(output_location + project_name + "_seqs.fasta",
+                  'w') as output_fh:
+            for counter, line in enumerate(sequences_fh):
+                if counter % 100000 == 0:
+                    print(counter)
+                if (line[0] != ">" and print_line):
+                    output_fh.write(line)
+                elif line[0] == ">":
+                    gi = line.split(" ")[1].split(":")[1]
+                    # name = " ".join(line.rstrip().split(" ")[2:])
+                    if gi in gis_to_taxids:
+                        taxid = str(gis_to_taxids[gi])
+                        taxid_to_increment = str(gis_to_taxids[gi])
+                        if truncation_level and\
+                                taxid_data[taxid][4] != truncation_level:
+                            for id in taxid_data[taxid][3]:
+                                if taxid_data[id][4] == truncation_level:
+                                    taxid = id
+                        if taxid_to_increment in taxid_counts:
+                            taxid_counts[taxid_to_increment] += 1
+                        else:
+                            taxid_counts[taxid_to_increment] = 1
+                        print_line = True
+                        output_fh.write(">" + gi + "-" + taxid + "\n")
+                    else:
+                        print_line = False
+                else:
+                    continue
+
+    # Updating gi counts in tree
+    print("Updating gi counts in tree")
+    for taxid in taxid_counts:
+        count = taxid_counts[taxid]
+        # print("Count : " + str(count))
+        taxid_data[taxid][5] += count
+        # print("Taxid : " + str(taxid))
+        for id in inclusion_tree[taxid][3]:
+            try:
+                taxid_data[id][5] += count
+                # print(str(id))
+            except IndexError:
+                print("id not in taxid_data : " + str(id))
+
+    print("Exiting output_sequences")
+
+
+def output_tree(taxid_data, inclusion_tree, output_location, project_name):
+    ##########################################################################
+    ###########################################################################
+    print("Entering output_tree")
+
+    print(len(inclusion_tree))
+
+    with open(output_location + project_name + ".tree", 'w') as output_fh:
+        for taxid in inclusion_tree:
+            print(taxid)
+            print(inclusion_tree[taxid][2])
+            print(inclusion_tree[taxid][5])
+            if inclusion_tree[taxid][2] == [''] and\
+                    inclusion_tree[taxid][5] != 0:
+                print("inside if")
+                output_string = ""
+                for id in taxid_data[taxid][3]:
+                    output_string += id + "\t" + taxid_data[id][
```

Comment:
Try string interpolation, should be more readable.

Also `output_strings` should become `output_lines`, then `'\n',join(output_lines)` at the end.